crates/compiler/gen_wasm/src/code_builder.rs

```diff
@@ -531,7 +529,7 @@ impl<'a> CodeBuilder<'a> {
         self.inst_base(opcode, pops, false);
 
         // We don't support block result types. Too hard to track types through arbitrary control flow.
-        self.code.push(BLOCK_NO_RESULT);
+        self.code.push(ValueType::VOID);
```

Comment:
what does this limitation mean in practice?

Reply:
The comment was probably written 6 months ago or more. It's about a (pretty minor) inefficiency in the code `gen_wasm` generates. We could emit fewer instructions if we did enough analysis and metadata tracking stuff. I tried many times to get that working, but it was extremely bug prone and not worth it.
Not worried about it though. The other day I noticed that in the WebAssembly version of the Roc compiler, Rust doesn't use block types either.
The change in this PR is just how to model this value in Rust code. It shouldn't be part of the `ValueType` enum but it's sort of related. It's like a `None` variant or something. The idea hit me to use an associated constant. Which made me feel like my Rust is better than it was back then!
Anyway it's a good question. I'll add to the comment to clarify it.

Reply:
I updated the comment.